File: android/src/main/java/com/getcapacitor/community/classes/options/InitializeOptions.java
package com.getcapacitor.community.classes.options;

import static com.getcapacitor.community.BridgefyHelper.makeBoolean;
import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import java.util.UUID;

public class InitializeOptions {

    @Nullable
    private UUID apiKey;

    @Nullable
    private Boolean verboseLogging;

    public InitializeOptions(PluginCall call) {
        @Nullable
        String apiKey = call.getString("apiKey");
        this.setAPIKey(apiKey);

        @Nullable
        Boolean verboseLogging = call.getBoolean("verboseLogging");
        this.setVerboseLogging(verboseLogging);
    }

    private void setAPIKey(@Nullable String apiKey) {
        this.apiKey = makeUUID(apiKey);
    }

    private void setVerboseLogging(@Nullable Boolean verboseLogging) {
        this.verboseLogging = verboseLogging; // makeBoolean(verboseLogging);
    }

    @Nullable
    public UUID getAPIKey() {
        return apiKey;
    }

    @Nullable
    public Boolean getVerboseLogging() {
        return verboseLogging;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Options/InitializeOptions.swift
import Foundation
import Capacitor

@objc public class InitializeOptions: NSObject {
    private var apiKey: UUID?
    private var verboseLogging: Bool?

    init(_ call: CAPPluginCall) {
        super.init()

        if let apiKey = call.getString("apiKey") {
            self.setAPIKey(apiKey)
        }

        if let verboseLogging = call.getString("verboseLogging") {
            self.setVerboseLogging(verboseLogging)
        }
    }

    func setAPIKey(_ apiKey: String?) {
        self.apiKey = Helper.makeUUID(apiKey)
    }
    func setVerboseLogging(_ verboseLogging: String?) {
        self.verboseLogging = Helper.makeBoolean(verboseLogging)
    }

    func getAPIKey() -> UUID? {
        return self.apiKey
    }
    func getVerboseLogging() -> Bool? {
        return self.verboseLogging
    }
}
